File: lib/bit-io/python/decode.py
#!/usr/bin/env python

from sys import *
import bit_io

if len(argv) != 3:
    stderr.write('Usage: {} INFILE OUTFILE\n'.format(argv[0]))
    exit(2)

with bit_io.BitReader(argv[1]) as input, open(argv[2], 'wb') as output:
    while True:
        c = input.readbits(7)
        if c is None: break
        output.write(chr(c))

File: lib/bit-io/ruby/decode.rb
#!/usr/bin/env ruby

require './bit_io.rb'

if ARGV.length != 2 then
  STDERR.print("Usage: #{$0} INFILE OUTFILE\n")
  exit 2
end

BitIO::BitReader.open(ARGV[0]) do |input|
  File.open(ARGV[1], 'wb') do |output|
    begin
      loop do
        output.write(input.readbits(7).chr)
      end
    rescue EOFError
      # expected
    end
  end
end
